pkg/github/github.go

```diff
@@ -146,6 +158,53 @@ func NewClient(ctx context.Context, token string, opts ...Option) *GitHubClient
 	return g
 }
 
+// ListRepositories all repositories and returns details about the repositories.
+func (g *GitHubClient) ListRepositories(ctx context.Context, owner string, opts *github.RepositoryListByOrgOptions) ([]*Repository, error) {
+	pageStart := func(i *int) bool { return i == nil }
+	pageEnd := func(i *int) bool { return *i == 0 }
+	uniqueResponses := make(map[int64]*Repository)
+	opt := *opts
+
+	var page *int // Use nil to indicate start of request.
+
+	for pageStart(page) || !pageEnd(page) {
```

Comment:
Why are we not paginating like the example: https://github.com/google/go-github?tab=readme-ov-file#pagination

This seems overly complex.

Reply:
Because we want to use the retry function to retry certain errors. It adds a little bit of complexity here.

Reply:
ack per offline comment